### PR TITLE
Add support for inlining functions (closes #1882)

### LINKED ISSUE #1882: extend inline transformation to support function calls
The inline transformation does not support function calls at the moment. This work will need #1880 to be on master first.

### PR DESCRIPTION
**Pull request by rupertford**

### CODE REVIEW

**Comment by rupertford** [comment]
Ready for first review, probably from @arporter or @sergisiso. This is a small change to the inline transformation to add support for inlining functions.

File: src/psyclone/tests/psyir/transformations/inline_trans_test.py [hunk 1]
```diff
@@ -617,6 +618,78 @@ def test_inline_non_local_import(fortran_reader, fortran_writer):
...
+    expected = (
+        "  subroutine run_it()\n"
+        "    real :: a\n"
+        "    real :: b\n"
+        "    real :: x\n\n"
+        "    x = 2.0\n"
+        "    a = x")
+    assert expected in output
+    assert Compile(tmpdir).string_compiles(output)
+
```
[the remark this answers is not in the document]
Comment (rupertford) [reply src/psyclone/tests/psyir/transformations/inline_trans_test.py:691]: Added.

File: src/psyclone/tests/psyir/transformations/inline_trans_test.py [hunk 2]
```diff
@@ -617,6 +618,78 @@ def test_inline_non_local_import(fortran_reader, fortran_writer):
...
+    psyir = fortran_reader.psyir_from_source(code)
+    routine = psyir.walk(Call)[0]
+    inline_trans = InlineTrans()
+    inline_trans.apply(routine)
+    output = fortran_writer(psyir)
+    expected = (
+        "  subroutine run_it()\n"
+        "    real :: a\n"
+        "    real :: b\n"
+        "    real :: func\n\n"
```
[the remark this answers is not in the document]
Comment (rupertford) [reply src/psyclone/tests/psyir/transformations/inline_trans_test.py:647]: We get invalid code :-). Fixed.

**Comment by rupertford** [comment]
Ready for next review from @arporter.

File: src/psyclone/psyir/transformations/inline_trans.py [hunk 3]
```diff
@@ -485,7 +491,9 @@ def _find_routine(call_node):
             definition cannot be found.
         '''
         name = call_node.routine.name
+        print(f"Looking for {name}")
```
Comment (arporter) [inline src/psyclone/psyir/transformations/inline_trans.py:494]: Please remove these prints.

**Review by arporter — CHANGES_REQUESTED** [review]
Requested changes have been made and I'm happy with the functionality now.
Tests and examples all compile OK.
No other changes have been made.
I would merge but there are two print statements that I think can be removed.
Once that's done I will merge...

File: src/psyclone/psyir/transformations/inline_trans.py (hunk 3 above)
Comment (rupertford) [reply src/psyclone/psyir/transformations/inline_trans.py]: Doh.

**Comment by rupertford** [comment]
Sorry about that. Should be OK now. Ready for next review from @arporter, GHA willing.

**Pull request merged**